### PR TITLE
Updated uperf workflow

### PR DESCRIPTION
**Pull request by jaredoconnell**
Now uses stop-if, as well as the workflow v0.2.0 format

This will be a draft until someone can confirm that it works, and until the uperf plugin is released. I'm having trouble with the service step in my setup.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jaredoconnell** [comment]
This is ready for review. It seems to occasionally fail, but I doubt that's because of the recent changes. It's possible that there's not enough delays or ready-checks as-designed. Especially related to the service.

File: network-performance/network-streaming-performance-k8s/workflow.yaml [hunk 1]
```diff
@@ -72,21 +71,27 @@ input:
             type_id: integer
 steps:
   uuidgen:
-    plugin: quay.io/arcalot/arcaflow-plugin-utilities:0.2.0
+    plugin:
+      src: quay.io/arcalot/arcaflow-plugin-utilities:0.2.0
```
Comment (dustinblack) [inline network-performance/network-streaming-performance-k8s/workflow.yaml:75]: Bump this to 0.4.0

File: network-performance/network-streaming-performance-k8s/workflow.yaml [hunk 2]
```diff
@@ -72,21 +71,27 @@ input:
...
+      deployment_type: image
     step: uuid
     input: {}
   kubeconfig:
-    plugin: quay.io/arcalot/arcaflow-plugin-kubeconfig:0.2.0
+    plugin:
+      src: quay.io/arcalot/arcaflow-plugin-kubeconfig:0.2.0
```
Comment (dustinblack) [inline network-performance/network-streaming-performance-k8s/workflow.yaml:81]: Bump this to 0.3.1

File: network-performance/network-streaming-performance-k8s/workflow.yaml [hunk 3]
```diff
@@ -185,9 +186,11 @@ steps:
               flowops:
                 - type: "disconnect"
   metadata:
-    plugin: quay.io/arcalot/arcaflow-plugin-metadata:0.1.0
+    plugin:
+      src: quay.io/arcalot/arcaflow-plugin-metadata:0.1.0
```
Comment (dustinblack) [inline network-performance/network-streaming-performance-k8s/workflow.yaml:190]: Bump this to 0.4.0

**Review by dustinblack — CHANGES_REQUESTED** [review]
I've noted a number of plugin version tags that should probably be updated. I will also try to find the time to functionally test this.

**Comment by jaredoconnell** [comment]
Versions bumped. Untested.

**Comment by dustinblack** [comment]
In functional testing on a kind cluster, something is keeping the uperf plugin from accessing the service:
```
2024-02-08T14:25:03+01:00	error	source=workflow	Output ID for step "uperf_client" is "error".
Output data: "map[error:Failed to parse output: could not find profile name.
Output: Error getting SSL CTX:1
Allocating shared memory of size 156624 bytes
Error connecting to arcaflow-svc-plugin-wzlzd

** TCP: Cannot connect to arcaflow-svc-plugin-wzlzd:20000 Connection refused
]"
```

Still looking into this.

**Review by dustinblack — APPROVED** [review]

**Pull request merged**